backends/qualcomm/passes/convert_scaled_dot_product_attention.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) Qualcomm Innovation Center, Inc.
+# All rights reserved
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import torch
+from executorch.exir.pass_base import ExportPass, PassResult
+from torch._decomp import get_decompositions
+from torch.fx.experimental.proxy_tensor import make_fx
+
+
+class ConvertScaledDotProductAttention(ExportPass):
```

Comment:
Why are you not doing this via graph rewrite? This kind of manual rewrite seems cumbersome. Also if this is for quantization purpose then we also have https://github.com/pytorch/pytorch/blob/main/test/quantization/pt2e/test_quantize_pt2e.py#L1255

Reply:
I’m sorry, I don’t quite understand. Could you please explain what you mean by graph rewrite? Because I want to decompose sdpa to annotate quantization_annotation for each node in sdpa, I refer the code in pytorch.
https://github.com/pytorch/pytorch/blob/13473df0d770e67f92cb4bb1b728f3746f160b7c/test/test_decomp.py#L941

Actually we run this pass intransform_for_annotation.
https://github.com/CodeLinaro/executorch/blob/2dc87fe8352299e77701ab8e9019201f545a9a1b/backends/qualcomm/qnn_quantizer.py#L322